api/client/client.go

```diff
@@ -108,6 +108,9 @@ type Client struct {
 	// closedFlag is set to indicate that the connection is closed.
 	// It's a pointer to allow the Client struct to be copied.
 	closedFlag *int32
+	// lastPing is a cached ping response.
+	lastPing   *proto.PingResponse
+	lastPingMu sync.Mutex
```

Comment:
We only cache Web API pings in our `lib/client`, and currently are hardly using our gRPC ping method unfortunately. I'll take it out of the PR for now but it adds a roundtrip to every request that requires MFA.

Reply:
There's probably still an opportunity for saving the pings in a single cache, regardless of source. A follow up seems best, since we are doing plenty over here. Thanks.